### PR TITLE
Added blogpost week 12

### PR DESCRIPTION
**Pull request by tvcastillod**
Hi, here are the blogpost for week 12.
image

### CODE REVIEW

File: docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
...ich must be square. The 6 images represent the faces of a cube". I had 4x15 coefficients and inside the function, a grid of RGB colors is made so then it can be mapped as a texture. To check if was passing the data correctly, I used the same value .5 for all the textures, so then I could pick a random texel get a specific color (gray), and pass it as *fragOutput0* to see if the value was correct. However, it didn't appear to work correctly as I couldn't get the expected color. To get the specific color I used `texture(sampler, P) <https://registry.khronos.org/OpenGL-Refpages/gl4/html/texture.xhtml>`_ which samples texels from the texture bound to sampler at texture coordinate P. Now, what I still need to figure out is which should be the corresponding texture coordinate, I have tried with random coordinates, since they are supposed to correspond to a point on the cube and since the information I have encoded in the texture is all the same I assumed that I would get the expected result for any set of values. It might be a problem with the data normalization, or maybe there is something failing on the texture definition. I need to review it in more detail to see where is the problem.
```
Comment (JoaoDell) [inline docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst:30]: * `Third, about passing the coefficients data through textures, I understand better (...)` 
* `I used the same value, .5, for all (...)`
* `(...) which should be the corresponding texture coordinate. I have tried with random coordinates, as they are (...)`
* `is all the same, I assumed (...)`
* `is something failing on the texture definition, but I need to review it (...)`

File: docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+There were different issues I needed to address for the ODF implementation. Even though I could not solve any of them completely, I did check each of the issues and made some progress. All the work in progress is being recorded in the following branch `SH-for-ODF-impl <https://github.com/tvcastillod/fury/tree/SH-for-ODF-impl>`_, which when ready will be associated with a well-structured PR.
+
+First, the scaling. For this part I was suggested to check Generalized Fractional Anisotropy **gfa** metric to adjust the scaling depending on the shape of the ODF glyph, i.e., the less the **gfa** the more sphere-shaped and smaller, so I had to associate a greater scaling for those. However, this did not work very well as I was unable to define an appropriate scale relation that would give an equitable result for each glyph. For this reason, I opted to use peak values which are extracted from the ODFs, setting the scales as 1/peak_value*0.4 and I got a more uniformly sized glyph without the need of setting it manually. That is a temporal solution as I would like to see better why this happens and if possible do the adjustment inside the shader instead of a precalculation.
```
Comment (JoaoDell) [inline docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst:14]: `First, about the scaling, I was suggested (...)`

File: docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Lastly, the colormapping. For this I created the texture based on a generic colormap from `matplotlib <https://matplotlib.org/stable/tutorials/colors/colormaps.html>`_. I was able to give some color to the glyph but it does not match correctly its shape. Some adjustment must be done regarding the texels, as the colormap is mapped on a cube. But I need it to fit the shape of the glyph correctly.
```
Comment (JoaoDell) [inline docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst:32]: * `Lastly, about the colormapping, I created (...)`
* `as the colormap is mapped on a cube, but I need it (...)`

File: docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+What is coming up next?
+-----------------------
+
+I will continue to explore more on how to handle textures so I can solve the issues related to the coefficient data and colormapping. Also, take a deeper look at the SH implementation and check what is the information needed to adjust the main direction of the ODF correctly.
+
+Did I get stuck anywhere?
+-------------------------
+
+As I mentioned I had some drawbacks in understanding the use of textures and how to retrieve the data inside the shaders. Is a topic that might take some time to manage properly but If I can master it and understand it better, it is a tool that can be useful later. Additionally, there are details of the SH implementation that I still need to understand and explore better in order to make sure I get exactly the same result as the current *odf_slicer* implementation.
```
Comment (JoaoDell) [inline docs/source/posts/2023/2023-08-24-week-12-tvcastillod.rst:46]: * `This is a topic that might (....)`
* `but if I can (...)`

**Review by JoaoDell — COMMENTED** [review]
Hey @tvcastillod, nice and detailed blogpost. It built successfully, and I just left some comments below, but everything is fine!

**Review by ganimtron-10 — COMMENTED** [review]
Hey @tvcastillod ,
I don't have any specific comments. @JoaoDell has covered most of it.

**Review by skoudoro — APPROVED** [review]
Nice blog post with great details. Thank you @tvcastillod, merging

**Pull request merged**